service/raft/group0_state_machine.cc

```diff
@@ -116,22 +117,25 @@ static bool should_flush_system_topology_after_applying(const mutation& mut, con
 }
 
 static future<> write_mutations_to_database(storage_proxy& proxy, gms::inet_address from, std::vector<canonical_mutation> cms) {
-    std::vector<mutation> mutations;
+    std::vector<std::pair<schema_ptr, frozen_mutation>> mutations;
     mutations.reserve(cms.size());
     bool need_system_topology_flush = false;
     try {
         for (auto& cm : cms) {
             auto& tbl = proxy.local_db().find_column_family(cm.column_family_id());
-            auto mut = co_await to_mutation_gently(cm, tbl.schema());
+            auto& s = tbl.schema();
+            auto mut = co_await to_mutation_gently(cm, s);
             need_system_topology_flush = need_system_topology_flush || should_flush_system_topology_after_applying(mut, proxy.data_dictionary());
-            mutations.emplace_back(std::move(mut));
+            mutations.emplace_back(s, co_await freeze_gently(mut));
         }
     } catch (replica::no_such_column_family& e) {
         slogger.error("Error while applying mutations from {}: {}", from, e);
         throw std::runtime_error(::format("Error while applying mutations: {}", e));
     }
 
-    co_await proxy.mutate_locally(std::move(mutations), tracing::trace_state_ptr());
+    co_await coroutine::parallel_for_each(mutations, [&] (auto& x) {
+        return proxy.mutate_locally(x.first, x.second, tracing::trace_state_ptr(), db::commitlog::force_sync::no);
+    });
```

Comment:
Maybe we should do this in storage_proxy(vector\<mutations>).

Reply:
can do, it makes sense

Reply:
done in v5